cfd_protocol/src/protocol/transactions.rs

```diff
@@ -366,13 +367,9 @@ pub fn punish_transaction(
         .context("commit transaction inputs != 1")?;
 
     let publish_them_sk = input
-        .witness
+        .extract_signatures()
         .iter()
-        .filter_map(|elem| {
-            let elem = elem.as_slice();
-            Signature::from_der(&elem[..elem.len() - 1]).ok()
-        })
-        .find_map(|sig| encsig.recover(SECP256K1, &sig, &pub_them_pk.key).ok())
+        .find_map(|sig| encsig.recover(SECP256K1, sig, &pub_them_pk.key).ok())
```

Comment:
Should we include this in the function as well? Basically making it a `extract_signature_by_key`? Otherwise we are more or less uselessly allocating now ...

Reply:
What about https://github.com/comit-network/hermes/pull/101/commits/d267cd6b641173d90ab39503e5e207f00acfc0b4?

Reply:
I didn't realize that we needed `encsig` for this functionality. Yeah that works!